atom/browser/api/atom_api_web_contents.cc

```diff
@@ -1457,14 +1457,23 @@ void WebContents::HasServiceWorker(const base::Callback<void(bool)>& callback) {
                      base::Unretained(wrapped_callback)));
 }
 
-void WebContents::UnregisterServiceWorker(
-    const base::Callback<void(bool)>& callback) {
+void OnUnregisterServiceWorker(scoped_refptr<util::Promise> promise,
+                               bool success) {
+  promise->Resolve(success);
+}
+
+v8::Local<v8::Promise> WebContents::UnregisterServiceWorker() {
+  scoped_refptr<util::Promise> promise = new util::Promise(isolate());
   auto* context = GetServiceWorkerContext(web_contents());
-  if (!context)
-    return;
+  if (!context) {
+    promise->RejectWithErrorMessage("Unable to get ServiceWorker context.");
+  }
```

Comment:
The old code doesn't invoke the callback if `context` can't be obtained. I'm 99% sure that's a bug in the old code, but pointing it out in case there's some crazy edge case that made that intentional